usaspending_api/transactions/delta_models/transaction_search.py

```diff
@@ -1,104 +1,491 @@
+    "pop_country_code": "STRING",
+    "pop_state_name": "STRING",
+    "pop_state_code": "STRING",
+    "pop_county_code": "STRING",
+    "pop_county_name": "STRING",
+    "pop_zip5": "STRING",
+    "pop_congressional_code": "STRING",
+    "pop_congressional_population": "INTEGER",
+    "pop_county_population": "INTEGER",
+    "pop_state_fips": "STRING",
+    "pop_state_population": "INTEGER",
+    "pop_city_name": "STRING",
+    "recipient_location_country_code": "STRING",
+    "recipient_location_country_name": "STRING",
+    "recipient_location_state_name": "STRING",
+    "recipient_location_state_code": "STRING",
+    "recipient_location_state_fips": "STRING",
+    "recipient_location_state_population": "INTEGER",
+    "recipient_location_county_code": "STRING",
+    "recipient_location_county_name": "STRING",
+    "recipient_location_county_population": "INTEGER",
+    "recipient_location_congressional_code": "STRING",
+    "recipient_location_congressional_population": "INTEGER",
+    "recipient_location_zip5": "STRING",
+    "recipient_location_city_name": "STRING",
+    "recipient_hash": "STRING",
+    "recipient_levels": "ARRAY<STRING>",
+    "recipient_name": "STRING",
+    "recipient_unique_id": "STRING",
+    "parent_recipient_hash": "STRING",
+    "parent_recipient_name": "STRING",
+    "parent_recipient_unique_id": "STRING",
+    "recipient_uei": "STRING",
+    "parent_uei": "STRING",
+    "awarding_toptier_agency_id": "INTEGER",
+    "funding_toptier_agency_id": "INTEGER",
+    "awarding_agency_id": "INTEGER",
+    "funding_agency_id": "INTEGER",
+    "awarding_toptier_agency_name": "STRING",
+    "funding_toptier_agency_name": "STRING",
+    "awarding_subtier_agency_name": "STRING",
+    "funding_subtier_agency_name": "STRING",
+    "awarding_toptier_agency_abbreviation": "STRING",
+    "funding_toptier_agency_abbreviation": "STRING",
+    "awarding_subtier_agency_abbreviation": "STRING",
+    "funding_subtier_agency_abbreviation": "STRING",
+    "treasury_account_identifiers": "ARRAY<INTEGER>",
+    "tas_paths": "ARRAY<STRING>",
+    "tas_components": "ARRAY<STRING>",
+    "federal_accounts": "STRING",
+    "disaster_emergency_fund_codes": "ARRAY<STRING>",
+    "awarding_office_code": "STRING",
+    "awarding_office_name": "STRING",
+    "funding_office_code": "STRING",
+    "funding_office_name": "STRING",
+}
+
+transaction_search_create_sql_string = fr"""
+    CREATE OR REPLACE TABLE {{DESTINATION_TABLE}} (
+        {", ".join([f'{key} {val}' for key, val in _transaction_search_types.items()])}
     )
     USING DELTA
-    LOCATION 's3a://{SPARK_S3_BUCKET}/{DELTA_LAKE_S3_PATH}/{DESTINATION_DATABASE}/{DESTINATION_TABLE}'
-    """
+    LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
+"""
+
+transaction_search_load_sql_string = fr"""
+    INSERT OVERWRITE rpt.transaction_search
+    (
+        {",".join([col for col in _transaction_search_types])}
+    )
+    SELECT
```

Comment:
I'll want to walk this through with you verbally, since there's so much packed into this matview-now-spark-sql statement. Things like:

- how `REGEXP_EXTRACT` worked out
- any array functions used
- e.g. whether `COLLECT_SET` is working okay as an array_agg substitute. Did we need to sort any of the agg'd results?
- UDFs working ok
- did we fix the `LATERAL JOIN` here
- if this was taken from the most recent transaction_search SQL, which incorporated all the UEI adjustments
- whether there's been any performance testing?
- any data validation testing thus far?

Reply:
Talked all these through. 
Actions:
- see if we can order the COLLECT_SET when aggregating a list of items, for future comparisons in MERGE's and the like